src/core/lib/iomgr/udp_server.cc

```diff
@@ -415,6 +404,30 @@ static int prepare_socket(grpc_socket_factory* socket_factory, int fd,
       gpr_log(GPR_INFO, "Failed to set socket overflow support");
     }
   }
+
+  if (so_reuseport && !grpc_is_unix_socket(addr) &&
```

Comment:
Previously, the block of code that called `bind_socket` was above `grpc_set_socket_sndbuf`.  I understand that you want to do `SO_REUSEPORT` before `bind_socket`  - but was there a reason you moved both setting SO_REUSEPORT and `bind_socket` below `grpc_set_socket_rcvbuf` ? 

Just curious

Reply:
I think in general it's better  to bind socket to the address after setting socket options. If setting socket options failed, there is no need to bind to that socket anymore.